### PR TITLE
Magn 9325: temporally bring node infront

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-9325](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9325) On hover over the node or compact view, the node and its preview bubble should be temporarily brought to the front of the view order

### Declarations

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs [hunk 1]
```diff
@@ -486,6 +495,24 @@ private void ExpandPreviewControl()
             }
         }
 
+        /// <summary>
+        /// Sets ZIndex of node the maximum value.
+        /// </summary>
+        private void BringInFront()
```
Comment (ramramps) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: BringToFront.

File: test/DynamoCoreWpfTests/NodeViewTests.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                Source = this,
+            });
+
+            // After click node should have The highest index.
+            Assert.AreEqual(nodeView.ViewModel.ZIndex, Dynamo.ViewModels.NodeViewModel.StaticZIndex);
+        }
+
+        [Test]
+        public void ZIndex_Test_MouseEnter_Leave()
+        {
```
Comment (ramramps) [inline test/DynamoCoreWpfTests/NodeViewTests.cs]: Good one.

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs [hunk 3]
```diff
@@ -486,6 +495,24 @@ private void ExpandPreviewControl()
...
+        {
+            if (IsMouseOver || PreviewControl.IsMouseOver || DynCmd.IsTestMode)
+            {
+                if (NodeViewModel.StaticZIndex == Int32.MaxValue)
+                {
+                    PrepareZIndex();
+                }
+                var index = ++NodeViewModel.StaticZIndex;
```
Comment (ramramps) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: I remember one of the test failed and you changed ++NodeViewModel.StaticZIndex  to  NodeViewModel.StaticZIndex++?

Comment (aosyatnik) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: No, it was NodeViewModel.StaticZIndex++ and I changed to ++NodeViewModel.StaticZIndex. It's line 324 in NodeView.xaml.cs : https://github.com/DynamoDS/Dynamo/pull/6091/files#diff-69be354bfe562b333ee34dec25c9bfd2L324

**Comment by ramramps** [comment]
Got it. LGTM

**Pull request merged**